core/server/common/src/test/java/alluxio/util/TarUtilsTest.java

```diff
@@ -78,6 +88,27 @@ public void nested() throws Exception {
     tarUntarTest(dir);
   }
 
+  @Test
+  public void testLargePosixGroupNumbers() throws Exception {
+    // when the TarArchiveEntry(File, String) constructor is called (as is used in
+    // TarUtils#writeTarGz), return a new instance of TarArchiveEntry that has a group id greater
+    // than the max id
+    Long groupId = TarArchiveEntry.MAXID + 100;
+    PowerMockito.whenNew(TarArchiveEntry.class)
+        .withParameterTypes(File.class, String.class)
+        .withArguments(any())
+        .thenAnswer(i -> {
+          TarArchiveEntry spy = PowerMockito.spy(
+              TarArchiveEntry.class.getConstructor(File.class, String.class)
+                  .newInstance(i.getArguments()));
+          PowerMockito.when(spy.getLongGroupId()).thenReturn(groupId);
```

Comment:
I think is uses the same code path, but just to be extra sure, maybe also add the mock for getLongUserId